<!--
  YOCS velocity smoother launcher
  -->

<launch>
  <arg name="node_name"             default="velocity_smoother"/>
  <arg name="nodelet_manager_name"  default="nodelet_manager"/>
  <arg name="config_file"           default="$(find ucar_controller)/param/vel_smooth.yaml"/>
  <!--<arg name="raw_cmd_vel_topic"     default="cmd_vel_teb"/>
  <arg name="smooth_cmd_vel_topic"  default="cmd_vel"/>
  <arg name="robot_cmd_vel_topic"   default="cmd_vel_pp"/>-->
    <arg name="raw_cmd_vel_topic"     default="cmd_vel_teb"/>
  <arg name="smooth_cmd_vel_topic"  default="cmd_vel"/>
  <arg name="robot_cmd_vel_topic"   default="cmd_vel_pp"/>
  <arg name="odom_topic"            default="odometry/filtered"/>

  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager"/>
  <node pkg="nodelet" type="nodelet" name="$(arg node_name)"
        args="load yocs_velocity_smoother/VelocitySmootherNodelet $(arg nodelet_manager_name)" >
        
    <!-- parameters -->
    <rosparam file="$(arg config_file)" command="load"/>

    <!-- velocity commands I/O -->
    <remap from="$(arg node_name)/raw_cmd_vel"    to="$(arg raw_cmd_vel_topic)"/>
    <remap from="$(arg node_name)/smooth_cmd_vel" to="$(arg smooth_cmd_vel_topic)"/>

    <!-- Robot velocity feedbacks -->
    <remap from="$(arg node_name)/robot_cmd_vel"  to="$(arg robot_cmd_vel_topic)"/>
    <remap from="$(arg node_name)/odometry"       to="$(arg odom_topic)"/>
  </node>
</launch>
